src/ser/mod.rs

```diff
@@ -204,3 +216,27 @@ where
         }),
     }
 }
+
+/// Serialize the given `T` as BSON bytes into the provided writer.
+#[inline]
+pub fn to_writer<T, W>(value: &T, mut writer: W) -> Result<()>
```

Comment:
This is equivalent to `writer.write_all(to_vec(value)?.as_slice())`, yes?  If so, I'm not sure it needs to be in the public API.